ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DefaultSynchronizer.java

```diff
@@ -249,6 +262,19 @@ private CompletableFuture<Void> handleSyncResult(final FastSyncState result) {
     }
   }
 
+  private void enableFallbackNodeFinder() {
+    if (worldStateStorage instanceof BonsaiWorldStateKeyValueStorage) {
```

Comment:
this one is not snapsync specific because I'm using GetNodeData and as fallback GetTrieNode. It can be used for fastsync and snapsync

In the future if we also have this kind of problem with forest I think it will be possible to adapt it